src/network.rs

```diff
@@ -0,0 +1,88 @@
+use errors::*;
+use std::path::{Path,PathBuf};
+use std::fs::{remove_file, rename};
+
+use args::Args;
+use fs;
+
+const BOOTPART_CONNECTIONS_DIR: &str = "system-connections";
+const NM_CONNECTIONS_DIR: &str = "/etc/NetworkManager/system-connections/";
+
+pub fn configure(args: &Args) -> Result<()> {
+    mirror_connections(args, BOOTPART_CONNECTIONS_DIR, NM_CONNECTIONS_DIR)?;
+    Ok(())
+}
+
+fn no_comments(s: &str) -> String {
+    let mut ret = String::new();
+    for line in s.lines() {
+        if ! line.starts_with("#") {
+            ret = ret + line + "\n";
+        }
+    }
+    ret
+}
+
+pub fn mirror_connections(args: &Args, bootpart_connections_dir: &str, nm_connections_dir: &str) -> Result<()> {
+
+    const IGNORE_EXT: &str = "ignore";
+    const CON_PERM: u32 = 600;
+    const HEADER: &str = "#
+# This system connection was mirrored to state partition
+#
+# If you want to force a new mirror operation on this file, rename this file removing the `.ignore`
+# extension and reboot your system.
+# This message can be left untouched as it will be handled by the OS.
+#
+";
+
+    let rootpart_con = Path::new(nm_connections_dir);
+
+    let bootpart = Path::new(&args.os_bootpart_path);
+    let bootpart_con_r = Path::new(bootpart_connections_dir);
+    let mut bootpart_con = PathBuf::from(bootpart);
+    bootpart_con.push(bootpart_con_r);
+
+    let filter_out_ext = vec![IGNORE_EXT, "tmp", "bkp"];
+
+    if ! bootpart_con.is_dir() {
+        return Ok(());
+    }
+
+    let entries = ::std::fs::read_dir(&bootpart_con)?;
+    for e in entries {
+	let e = e?;
+        if e.path().is_dir() {
+            continue;
+        }
+        if let Some(ext) = e.path().extension() {
```

Comment:
We can have a function similar to our `fs::file_name` for getting the extension of a file.